packages/tasit-action/src/contract/Action.js

```diff
@@ -180,16 +180,29 @@ export class Action extends Subscription {
       }
     };
 
-    let ethersListener;
-
-    if (once) {
-      ethersListener = async blockNumber => {
-        await baseEthersListener(blockNumber);
-        this.off(eventName);
-      };
-    } else {
-      ethersListener = baseEthersListener;
-    }
+    // Note:
+    // On the development env (using ganache-cli)
+    // Blocks are being mined simultaneously and generating a sort of unexpected behaviors like:
```

Comment:
Maybe if we pick a very, very short but nonzero block time it would help?

Reply:
I've tried that but not with extremely low block time. I'll make a try.

Reply:
I've tried these blockTimes: 0.001, 0.01 and 0.1. The lowest values increased the occurrences of the parallel calls.
I guess that could be related to auto block generation for each transaction by `ganache-cli` and maybe this feature could help (https://github.com/trufflesuite/ganache-core/issues/248).
I've open this issue to track that: https://github.com/tasitlabs/TasitSDK/issues/380

Reply:
Yep, agreed that that is the most likely cause. I agree that that Ganache feature could help. I added a comment on that issue.

Is the non-determinism mostly for tests where the listener is added before or after sending the tx? Or is it a little of both?

We're sure it's not just getting called again with another confirmation for each instantaneous block, right? If so, we could probably disable that in our code and check number of confirmations on each listener call. Or something like that.

Reply:
Feel free to move my comment over to that issue instead

Reply:
If we move it there, we can resolve this thread. Thanks for testing those block times!

Reply:
Comment extracted to issue. Resolving this.